src/pages/EnvironmentsPage/environmentsPermissions.ts

```diff
@@ -0,0 +1,26 @@
+import { EnvironmentPermissionsProvider } from 'src/analysis/Environments/Environments';
+import { getCreatorForCompute } from 'src/analysis/utils/resource-utils';
+import { PersistentDisk } from 'src/libs/ajax/leonardo/models/disk-models';
+import { getTerraUser } from 'src/libs/state';
+
+const makePermissionsProvider = (userEmailGetter: () => string): EnvironmentPermissionsProvider => {
+  const permissionsProvider: EnvironmentPermissionsProvider = {
+    canDeleteDisk: (disk: PersistentDisk) => {
+      const currentUserEmail = userEmailGetter();
+      const canDelete = disk.auditInfo.creator === currentUserEmail;
+      return canDelete;
+    },
+    canPauseResource: (resource) => {
```

Comment:
It could be worth typing this here as `App | PersistentDisk | Runtime`. Or making a `Resource` type that is the union of the three